### PR TITLE
[Proposal] Fungible token standard with approval

### PR DESCRIPTION
**Pull request by baerae-colin**
This token standard is compatible with ERC-20 standard of Ethereum.

### CODE REVIEW

**Comment by aidan-kwon** [comment]
@kjhman21 By the way, Klyatn has an implementation of the enhanced ERC-20 standard in https://github.com/klaytn/klaytn/blob/dev/contracts/sc_erc20/sc_token.sol. Will it be a new KIP?

**Comment by baerae-colin** [comment]
@aidan-kwon 

> Will it be a new KIP?

Yes. It should be.

**Review by jimni1222 — APPROVED** [review]

**Comment by KimKyungup** [comment]
Could you tell me why you propose a fungible toke standard and one more fungible token standard with approval? Do we have some reason to separate them?

**Comment by aidan-kwon** [comment]
I think `transfer` should be placed in `Summary of Methods and Events`.  As far as I know, the function is one of the most important logic enabling fungible tokens to be transferred from a user to another user. 

By the way, we might declare the safe version of ERC-20 such as safeERC20 as a standard.

**Comment by baerae-colin** [comment]
@KimKyungup 
> Could you tell me why you propose a fungible toke standard and one more fungible token standard with approval? Do we have some reason to separate them?

1. approval function has a security caveat as described in ERC-20.
> NOTE: To prevent attack vectors like the one described here and discussed here, clients SHOULD make sure to create user interfaces in such a way that they set the allowance first to 0 before setting it to another value for the same spender. THOUGH The contract itself shouldn't enforce it, to allow backwards compatibility with contracts deployed before

2. I don't think approval function is a basic function for fungible tokens.

**Comment by KimKyungup** [comment]
@kjhman21 
I understood the security caveat. But it is being used with ERC-20.
I don't know why we should make a token standard without `approval function`.
Because most people think fungible token has approval function. They might try to use it later.

If we want to do something, why don't we add a new standard (safeERC20) as @aidan-kwon said.

**Comment by baerae-colin** [comment]
@KimKyungup Actually it cannot be solved in safe way theoretically. The solution provided by ERC20 is practical such as approve(0) should be executed prior to update allowance.

And also, we will have the token standard as the same as ERC20. This can be another token standard.

**Review by KimKyungup — APPROVED** [review]
This proposal is same with ERC20.
It's ok. I will survey more about ERC20 without approval.

**Review by aidan-kwon — APPROVED** [review]

File: KIPs/kip-fungible_token_with_approval.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+---
+kip: <to be assigned>
+title: Fungible Token Standard With Approval
+author: Junghyun Colin Kim <colin.kim@groundx.xyz>
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category (*only required for Standard Track): Token
```
Comment (baerae-colin) [inline KIPs/kip-fungible_token_with_approval.md]:
```suggestion
category: Token
```

File: KIPs/kip-fungible_token_with_approval.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+---
+kip: <to be assigned>
```
Comment (baerae-colin) [inline KIPs/kip-fungible_token_with_approval.md]:
```suggestion
kip: 5
```

**Comment by baerae-colin** [comment]
Since this PR makes too many token standards for fungible tokens, it is closed. We will define a unified fungible token standard for Klaytn.

**Pull request closed without merging**